template/lib.typ

```diff
@@ -136,15 +159,25 @@
   )
 }
 
-#let index() = context {
-    [= Índice]
+#let index(target) = context {
```

Comment:
I was thinking that we could make `target` a selector instead, like what it is in `outline`: https://typst.app/docs/reference/model/outline/
That is, an expression that is directly passed onto `query` for obtaining relevant elements. That way we can pass things like `index(target: heading.where(level: 1))`
As for the title, we can have it also be a parameter. Both parameters should be optional and have a default value for regular headings. Same thing as `outline`, to be as consistent as possible.

Reply:
I think we should talk about how you want the index. There are some functionalities that I dont know how to implement

Reply:
The index function is done. Tell me if now it suits the idea that you had in mind